### ISSUE TITLE
add image= option to /share

### ISSUE THREAD

**Issue by naknomum** [mastodon/mastodon#24215]
### Pitch

right now, it appears there are `title=`, `text=`, and `url=` options. (although they do not seem to do much, other than be concatenate those values together.)  it would be nice to have an `image=IMAGE_URL` option, which would attach the image to the post (like a thumbnail or preview).

### Motivation

right now, any off-site embedded "share on mastodon" widget will produce a text-only post.  this option would allow for an optional visual element to be in the post.

**Comment by andypiper** [comment]
This is technically more complicated to do than including text or URL in a shared post, because the image would need to be uploaded to the Mastodon instance.

**Comment by naknomum** [comment]
yeah, plenty of technical weirdness around this for sure. there would have to be an initial process which grabbed the image from the url and created it on the mastodon instance first. would have to handle bad urls, maybe `data:` urls, and the like.  doable, but not simple, for sure.  :slightly_smiling_face:
